src/physics/cam/co2_data_flux.F90

```diff
@@ -1,164 +1,225 @@
+      integer           :: istat
+      integer           :: rc
+      character(len=*), parameter :: subname = 'co2_data_flux_init'
+      !--------------------------------------------
+
+      ! Allocate data_flux_fuel%co2flx
+      allocate( data_flux_fuel%co2flx(pcols,begchunk:endchunk), stat=istat)
+      call alloc_err(istat, subname, 'data_flux_fuel%co2flx', pcols*(endchunk-begchunk+1))
+
+      ! Initialize sdat_co2
+      call shr_strdata_init_from_inline(sdat_co2,             &
+           my_task             = iam,                         &
+           logunit             = iulog,                       &
+           compname            = 'ATM',                       &
+           model_clock         = model_clock,                 &
+           model_mesh          = model_mesh,                  &
+           stream_meshfile     = trim(co2flux_fuel_meshfile), &
+           stream_filenames    = (/co2flux_fuel_datafile/),   &
+           stream_yearFirst    = co2flux_fuel_year_first,     &
+           stream_yearLast     = co2flux_fuel_year_last,      &
+           stream_yearAlign    = co2flux_fuel_year_align,     &
+           stream_fldlistFile  = (/data_flux_fuel%varname/),  &
+           stream_fldListModel = (/data_flux_fuel%varname/),  &
+           stream_lev_dimname  = 'null',                      &
+           stream_mapalgo      = 'consf',                     &
+           stream_offset       = 0,                           &
+           stream_taxmode      = trim(co2flux_fuel_taxmode),  &
+           stream_dtlimit      = 1.0e30_r8,                   &
+           stream_tintalgo     = 'linear',                    &
+           stream_name         = 'CO2 forcing data ',         &
+           rc                  = rc)
+      call chkrc(rc,__LINE__,u_FILE_u)
+
+   end subroutine co2_data_flux_init
+
+   !===============================================================================
+   subroutine co2_data_flux_advance()
+
+      !-------------------------------------------------------------------------------
+      ! Advance the contents of a co2_data_flux_type sdat (map and interpolate in time)
+      !-------------------------------------------------------------------------------
+
+      use dshr_methods_mod , only : dshr_fldbun_getfldptr
+      use dshr_strdata_mod , only : shr_strdata_advance
+      use ppgrid           , only : begchunk, endchunk
+      use phys_grid        , only : get_ncols_p
+      use time_manager     , only : get_curr_date
+      use cam_esmf_mod     , only : cam_esmf_global_sum
+
+      ! Local variables
+      integer :: icol,lchnk,g
+      integer :: year    ! year (0, ...) for nstep+1
+      integer :: mon     ! month (1, ..., 12) for nstep+1
+      integer :: day     ! day of month (1, ..., 31) for nstep+1
+      integer :: sec     ! seconds into current date for nstep+1
+      integer :: mcdate  ! Current model date (yyyymmdd)
+      integer :: rc
+      logical :: first_time = .true.
```

Comment:
This silently adds the `save` attribute to the subroutine which we do not want. Please move this to the module scope (e.g., as `first_advance_call`).